src/dmn/trn_transit_xdp_usr.c

```diff
@@ -220,6 +220,21 @@ int trn_update_network(struct user_metadata_t *md, struct network_key_t *netkey,
 		TRN_LOG_ERROR("Store network mapping failed (err:%d)", err);
 		return 1;
 	}
+
+	if (md->xdp_flags == XDP_FLAGS_HW_MODE){
+		struct network_t_offload net_offload;
+		net_offload.prefixlen = net->prefixlen;
+		memcpy(net_offload.nip, net->nip, sizeof(net_offload.nip));
+		net_offload.nswitches = net->nswitches;
```

Comment:
What happens if net->nswitches > TRAN_MAX_NSWITCH_OFFLOAD?

Reply:
Due to the size limitation of key+value, we can not add all switches_ips into net_offload.
Thus, we just copy a few switches_ips into net_offload.
I think one reasonable way is to classify switches into two types, i.e., supporting offload or not supporting, but this may needs discussions.

Reply:
Then we should add a check here and fail the request. It is better to catch this early in order to avoid unpredictable errors.